firebase-perf/src/test/java/com/google/firebase/perf/util/FirstDrawDoneListenerTest.java

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.firebase.perf.util;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import android.os.Handler;
+import android.view.View;
+import android.view.ViewTreeObserver;
+import androidx.test.core.app.ApplicationProvider;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+/** Unit tests for {@link FirstDrawDoneListener}. */
+@RunWith(RobolectricTestRunner.class)
+public class FirstDrawDoneListenerTest {
+  @Mock private Handler handler;
+  @Captor private ArgumentCaptor<Runnable> callbackCaptor;
+
+  private View testView;
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+    testView = new View(ApplicationProvider.getApplicationContext());
+  }
+
+  @Test
+  @Config(sdk = 25)
+  public void registerForNextDraw_delaysAddingListenerForAPIsBelow26()
+      throws NoSuchFieldException, IllegalAccessException {
+    // Add a listener first so ViewTreeObserver.mOnDrawListeners is initialized and non-null
+    testView.getViewTreeObserver().addOnDrawListener(() -> {});
+    Field mOnDrawListeners =
+        android.view.ViewTreeObserver.class.getDeclaredField("mOnDrawListeners");
+    mOnDrawListeners.setAccessible(true);
+    ArrayList<ViewTreeObserver.OnDrawListener> listOfListeners =
+        (ArrayList<ViewTreeObserver.OnDrawListener>)
+            mOnDrawListeners.get(testView.getViewTreeObserver());
+    assertThat(listOfListeners).isNotNull();
+    assertThat(listOfListeners.size()).isEqualTo(1);
+
+    // OnDrawListener is not registered, it is delayed for later
```

Comment:
I don't see a way to do it because `View` class doesn't have public `View.dispatchAttachView()` unlike the other listeners we have for `ViewTreeObserver` (`ViewTreeObserver.dispatchOnDraw()` and `ViewTreeObserver.dispatchOnGlobalLayout()`), and although there's a private `View.dispatchAttachView()` it uses parameter classes that are also not visible to us.

Reply:
Figured out a way to do it through reflection, PTAL @visumickey @raymondlam